src/index.ts

```diff
@@ -1571,9 +1569,10 @@ export class BigQuery extends common.Service {
     optionsOrCallback?:
       | QueryOptions
       | SimpleQueryRowsCallback
+      | RowsCallback
       | QueryRowsCallback,
-    cb?: SimpleQueryRowsCallback | QueryRowsCallback
-  ): void | Promise<SimpleQueryRowsResponse> | Promise<QueryRowsResponse> {
+    cb?: SimpleQueryRowsCallback | RowsCallback
```

Comment:
It surprised me that `QueryRowsCallback` had to change `RowsCallback`, since in the case of model queries we still return the original response:

```
 job!.getQueryResults(options, callback as QueryRowsCallback);
```

☝️ this is the original slow path right? which we returned as a `QueryRowsCallback`?

Mainly just making sure code-paths return the same shaped object.